### PR TITLE
Components Testing

### PR DESCRIPTION
**Pull request by DarkxDev**
- Set up React Testing Library.
- Wrote unit tests for all React components.
- Used Jest snapshots to test the components.

### CODE REVIEW

File: src/logic/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import operate from './operate';
+
+test('Add funcionality test', () => {
+  expect(operate(2, 3, '+')).toBe('5');
+});
```
Comment (vickymarz) [inline src/logic/operate.test.js:5]: - [ ] Kindly ensure that all the functionalities & scenarios of the operate.js are covered in your test such as division, subtraction, multiplication, and modulus.

you only have a test for addition, so kindly add test for the other functionality i mentioned above 👍🏽

File: src/logic/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    next: 5,
+    operation: '+',
+  };
+  const output = {
+    total: null,
+    next: null,
+    operation: null,
+  };
+  expect(calculate(input, buttonName)).toEqual(output);
+});
```
Comment (vickymarz) [inline src/logic/calculate.test.js:16]: - [ ] Kindly ensure that all the functionalities & scenarios of the calculate.js are covered in your test such as division, subtraction, multiplication, and modulus.

you only have a test for addition, so kindly add test for the other functionality i mentioned above 👍🏽

**Review by vickymarz — CHANGES_REQUESTED** [review]
Hi @banzaman
You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- All tests are passing ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @DarkxDev , 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔
- No linter errors ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**